### PR TITLE
add default dynamic nfs client analyzer

### PR DESCRIPTION
**Pull request by cbodonnell**
#### What type of PR is this?

type::feature

#### What this PR does / why we need it:

This PR adds a default dynamic analyzer to KOTS that will run when snapshots are enabled for the license.  This will help to reduce the support burden for issues where customers cannot configure NFS snapshots because an NFS client package is not installed on the host(s).

**Fail:**
nfs-client-analyzer-fail

**Pass:**
nfs-client-analyzer-pass

#### Which issue(s) this PR fixes:

Fixes [SC-45145](https://app.shortcut.com/replicated/story/45145/analyzer-missing-nfs-client-package)

#### Special notes for your reviewer:

This will only produce a failure if snapshots are enabled and the user has tried to configure NFS snapshots without a client package installed.  It could also potentially produce a failure if the user has resolved the issue, but the message we search for still exists in the event logs.

This was added as a default dynamic analyzer so that we can include it only for snapshot-enabled licenses and also target the desired namespace when searching the event logs.

#### Does this PR introduce a user-facing change?

```release-note
Support bundles generated in the "Troubleshoot" tab will now indicate if an NFS client package was missing when configuring snapshots.
```

#### Does this PR require documentation?

NONE

### CODE REVIEW

File: pkg/supportbundle/spec.go [hunk 1]
```diff
@@ -527,6 +527,36 @@ func getDefaultDynamicAnalyzers(app *apptypes.App) []*troubleshootv1beta2.Analyz
 	analyzers := make([]*troubleshootv1beta2.Analyze, 0)
 	analyzers = append(analyzers, makeAPIReplicaAnalyzer())
 
+	license, err := store.GetStore().GetLicenseForAppVersion(app.ID, app.CurrentSequence)
```
Comment (sgalsaleh) [inline pkg/supportbundle/spec.go:530]:
```suggestion
	license, err := store.GetStore().GetLatestLicenseForApp(app.ID)
```

File: pkg/supportbundle/spec.go [hunk 2]
```diff
@@ -527,6 +527,36 @@ func getDefaultDynamicAnalyzers(app *apptypes.App) []*troubleshootv1beta2.Analyz
...
+	if err != nil {
+		logger.Errorf("Failed to get license for dynamic analyzers for app id %s sequence %d: %v", app.ID, app.CurrentSequence, err)
```
Comment (sgalsaleh) [inline pkg/supportbundle/spec.go:532]:
```suggestion
		logger.Errorf("Failed to get latest license for app %s: %v", app.Slug, err)
```

File: pkg/supportbundle/spec.go [hunk 3]
```diff
@@ -527,6 +527,36 @@ func getDefaultDynamicAnalyzers(app *apptypes.App) []*troubleshootv1beta2.Analyz
...
+			TextAnalyze: &troubleshootv1beta2.TextAnalyze{
+				AnalyzeMeta: troubleshootv1beta2.AnalyzeMeta{
+					CheckName: "NFS Client Package",
+				},
+				FileName:     fmt.Sprintf("cluster-resources/events/%s.json", util.PodNamespace),
+				RegexPattern: "bad option; for several filesystems \\(e\\.g\\. nfs, cifs\\) you might need a \\/sbin\\/mount\\..+type.+ helper program\\.",
+				Outcomes: []*troubleshootv1beta2.Outcome{
+					{
+						Fail: &troubleshootv1beta2.SingleOutcome{
+							When:    "true",
```
Comment (sgalsaleh) [inline pkg/supportbundle/spec.go:544]: i don't think this is needed

File: pkg/supportbundle/spec.go [hunk 4]
```diff
@@ -527,6 +527,36 @@ func getDefaultDynamicAnalyzers(app *apptypes.App) []*troubleshootv1beta2.Analyz
...
+							Message: "NFS client errors were found. Refer to [documentation](https://docs.replicated.com/enterprise/snapshots-configuring-nfs) on how to configure NFS snapshots.",
+							URI:     "https://docs.replicated.com/enterprise/snapshots-configuring-nfs",
+						},
+					},
+					{
+						Pass: &troubleshootv1beta2.SingleOutcome{
+							When:    "false",
```
Comment (sgalsaleh) [inline pkg/supportbundle/spec.go:551]: this means that a success outcome won't ever show up, no?

**Review by sgalsaleh — APPROVED** [review]

**Review by divolgin — APPROVED** [review]

**Pull request closed without merging**